src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -1602,12 +1626,16 @@ public function getParameter(string $name): array|bool|string|int|float|\UnitEnu
         }
 
         if (!(isset($this->parameters[$name]) || isset($this->loadedDynamicParameters[$name]) || \array_key_exists($name, $this->parameters))) {
-            throw new ParameterNotFoundException($name);
+            throw new ParameterNotFoundException($name, extraMessage: self::NONEMPTY_PARAMETERS[$name] ?? null);
         }
         if (isset($this->loadedDynamicParameters[$name])) {
             return $this->loadedDynamicParameters[$name] ? $this->dynamicParameters[$name] : $this->getDynamicParameter($name);
         }
 
+        if (isset(self::NONEMPTY_PARAMETERS[$name]) && (null === $this->parameters[$name] || '' === $this->parameters[$name])) {
+            throw new \Symfony\Component\DependencyInjection\Exception\EmptyParameterValueException(self::NONEMPTY_PARAMETERS[$name]);
+        }
+
```

Comment:
[todo, self-note] this doesn't take the dynamic parameters into account (previous ‘if’ block). Needs refactoring.

Reply:
Updated